### PR TITLE
Update Go Dependencies

### PR DESCRIPTION
**Pull request by xrstf**
**What this PR does / why we need it**:
This bumps all direct dependencies to their latest versions.

**Optional Release Note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/cloudprovider/provider/linode/provider.go [hunk 1]
```diff
@@ -109,8 +109,7 @@ func getClient(token string) linodego.Client {
 	oauthClient := oauth2.NewClient(context.Background(), tokenSource)
 
 	client := linodego.NewClient(oauthClient)
-	ua := fmt.Sprintf("Kubermatic linodego/%s", linodego.Version)
-	client.SetUserAgent(ua)
+	client.SetUserAgent("Kubermatic linodego")
```
Comment (xrstf) [inline pkg/cloudprovider/provider/linode/provider.go:112]: Sadly, because of https://github.com/linode/linodego/pull/156

**Comment by moadqassem** [comment]
@xrstf can you rebase your PR :)

**Pull request merged**